kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/HttpClientUtils.java

```diff
@@ -183,6 +184,16 @@ private static OkHttpClient createHttpClient(final Config config, final Consumer
                 additionalConfig.accept(httpClientBuilder);
             }
 
+          if (config.getCustomHeaders() != null && !config.getCustomHeaders().isEmpty()) {
+            httpClientBuilder.addNetworkInterceptor(chain -> {
```

Comment:
I think it should be `httpClientBuilder.addInterceptor(`, see line 115 where some other `Interceptors` are added.

Reply:
No, both `networkInterceptor` and `Interceptor` are basically a `List<Interceptor>`. Seems like there was some problem in my setup due to which I couldn't see headers being applied. Now I can see them. Sorry for confusion :laughing: